File: src/EAM.Shared/DTOs/AgentDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using EAM.Shared.Enums;

namespace EAM.Shared.DTOs;

public class AgentDto
{
    public Guid Id { get; set; }
    
    [Required]
    [MaxLength(255)]
    public string MachineId { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(255)]
    public string MachineName { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(255)]
    public string UserName { get; set; } = string.Empty;
    
    [MaxLength(255)]
    public string? OsVersion { get; set; }
    
    [MaxLength(50)]
    public string? AgentVersion { get; set; }
    
    [JsonPropertyName("last_seen")]
    public DateTime LastSeen { get; set; } = DateTime.UtcNow;
    
    public AgentStatus Status { get; set; } = AgentStatus.Active;
    
    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: src/app/shared/models/agent-dto.ts
export interface AgentDto {
  id: string;
  machineId: string;
  machineName: string;
  userName: string;
  osVersion?: string;
  agentVersion?: string;
  lastSeen: Date;
  status: AgentStatus;
  createdAt: Date;
  updatedAt: Date;
}

export interface AgentResponse {
  agents: AgentDto[];
  totalCount: number;
  pageSize: number;
  currentPage: number;
  totalPages: number;
}

export interface AgentFilters {
  machineId?: string;
  machineName?: string;
  userName?: string;
  status?: AgentStatus;
  pageSize?: number;
  currentPage?: number;
}

export enum AgentStatus {
  Active = 'Active',
  Inactive = 'Inactive',
  Offline = 'Offline',
  Maintenance = 'Maintenance',
  Disabled = 'Disabled'
}

export interface AgentMetrics {
  totalAgents: number;
  activeAgents: number;
  offlineAgents: number;
  inactiveAgents: number;
  maintenanceAgents: number;
  disabledAgents: number;
}

export interface AgentHeartbeat {
  agentId: string;
  timestamp: Date;
  systemInfo?: {
    cpuUsage?: number;
    memoryUsage?: number;
    diskUsage?: number;
  };
}